src/main/java/com/arpnetworking/tsdcore/sinks/PeriodicStatisticsSink.java

```diff
@@ -401,11 +435,30 @@ protected Builder self() {
             return this;
         }
 
+        private boolean validate(final ImmutableMap<String, String> ignored) {
+            final Set<String> mappedTargetDimensions = Sets.newHashSet(_mappedDimensions.values());
+
+            // Assert that no two mapped dimensions target the same value
+            if (mappedTargetDimensions.size() != _mappedDimensions.size()) {
+                return false;
+            }
+
+            // Assert that no mapped dimension target is also a dimension
```

Comment:
Would it be correct to rename a dimension and subsequently replace it?

e.g.
```
dimensions = {"a", "b"}
mappedDimensions = {"b": "c", "a" : "b"}
```
We replace `b` with `c`, and then `a` is mapped to `b` in the output. Because input `b` was itself mapped, there is no conflict. I admit that this is a bit contrived but it is a situation that doesn't have a conflict so I would expect it to work.

Reply:
The example you provided is invalid because `b` is mapped to `b` (via `dimensions) and `a` is also mapped to `b` (with value from `a`; via `mappedDimensions`) so there is an output conflict here.

I describe a case below where you have no effectively conflict; say if your example were:
```
dimensions = {"a", "b"}
mappedDimensions = {"b": "c", "b" : "b"}
```

Here b is in both lists as a target (detected as a conflict) but the effect is a no-op since it would end up having the same value (input `b`) in both cases.